src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    }
+
+    let string = '';
+
+    for (let i = 0; i < this.length; i++) {
```

Comment:
Don't you think first `if`  is extra here? 
Certainly, it is more of a question of clearness of code, but if you delete it, function should work just fine.
Your loop will not start ( `0 < (this.length = 0) => false`)  and `string = ''` will be returned.